pkg/stub/handler.go

```diff
@@ -99,11 +103,90 @@ type Handler struct {
 	skippedTemplates    map[string]bool
 	skippedImagestreams map[string]bool
 
+	imagestreamFile map[string]string
+	templateFile    map[string]string
+
 	deleteInProgress bool
 
 	mutex *sync.Mutex
 }
 
+func (h *Handler) prepWatchEvent(name string, annotations map[string]string) (*v1alpha1.SamplesResource, string, bool) {
+	if h.deleteInProgress {
+		return nil, "", false
+	}
+	srcfg, err := h.sdkwrapper.Get(v1alpha1.SamplesResourceName)
+	if srcfg == nil || err != nil {
+		logrus.Printf("Received watch event %s but ignoring since not have the SamplesResource yet: %#v %#v", name, err, srcfg)
+		return nil, "", false
+	}
+	proceed, err := h.ProcessManagementField(srcfg)
+	if !proceed || err != nil {
+		return nil, "", false
+	}
+
+	cm, err := h.configmapclientwrapper.Get(h.namespace, v1alpha1.SamplesResourceName)
+	if err != nil {
+		logrus.Warningf("Problem accessing config map during image stream watch event processing: %#v", err)
+		return nil, "", false
+	}
+	if annotations != nil {
+		isv, ok := annotations[v1alpha1.SamplesVersionAnnotation]
+		if ok && isv == v1alpha1.CodeLevel {
+			return nil, "", false
+		}
+	}
+	filePath, ok := cm.Data[name]
```

Comment:
this doesn't handle the possibility of a template + imagestream w/ the same name...you need to prefix your keys w/ the type.  

I also think you should prefix them with the namespace and make sure that the event you're processing is for a template/imagestream from the openshift namespace unless you've proven to yourself that the SDK watcher actually respects the namespace you asked to watch (i'm pretty sure it doesn't).